crates/red_knot_python_semantic/src/types/mro.rs

```diff
@@ -362,15 +365,40 @@ impl<'db> ClassBase<'db> {
                 | KnownInstanceType::Union
                 | KnownInstanceType::NoReturn
                 | KnownInstanceType::Never
+                | KnownInstanceType::Final
+                | KnownInstanceType::NotRequired
+                | KnownInstanceType::TypeGuard
+                | KnownInstanceType::TypeIs
+                | KnownInstanceType::TypingSelf
+                | KnownInstanceType::Unpack
+                | KnownInstanceType::ClassVar
+                | KnownInstanceType::Concatenate
+                | KnownInstanceType::Required
+                | KnownInstanceType::TypeAlias
+                | KnownInstanceType::ReadOnly
                 | KnownInstanceType::Optional => None,
                 KnownInstanceType::Any => Some(Self::Any),
                 // TODO: Classes inheriting from `typing.Type` et al. also have `Generic` in their MRO
-                KnownInstanceType::Type => {
-                    ClassBase::try_from_ty(db, KnownClass::Type.to_class_literal(db))
-                }
-                KnownInstanceType::Tuple => {
-                    ClassBase::try_from_ty(db, KnownClass::Tuple.to_class_literal(db))
-                }
+                KnownInstanceType::Dict
+                | KnownInstanceType::List
+                | KnownInstanceType::Tuple
+                | KnownInstanceType::FrozenSet
+                | KnownInstanceType::Type
+                | KnownInstanceType::Set => Self::try_from_ty(
+                    db,
+                    builtins_symbol(db, &known_instance.as_str().to_lowercase())
```

Comment:
Hmmm... ok :) I guess this is perfectly accurate, but I'm tempted to say we should make the mapping explicit.

Reply:
ugh fiiiiiine

Reply:
I'm also totally fine with not doing it, if you don't think it's worth it :)

Reply:
nah I think you're right that this is probably "more correct", I was just being lazy 😆